src/main/java/de/tum/in/www1/artemis/service/export/DataExportQuizExerciseCreationService.java

```diff
@@ -48,22 +67,30 @@ public DataExportQuizExerciseCreationService(QuizSubmissionRepository quizSubmis
      * @param participation  the participation for which the export should be created
      * @param outputDir      the directory in which the export should be stored
      * @param includeResults true if the results should be included in the export (if the due date is over)
-     * @throws IOException if an error occurs while accessing the file system.
+     * @param exportErrors   an optional list of errors that occurred during the export
+     * @return true if the export was successful, false otherwise
      */
-    public void createQuizAnswersExport(QuizExercise quizExercise, StudentParticipation participation, Path outputDir, boolean includeResults) throws IOException {
+    private boolean createQuizAnswersExport(QuizExercise quizExercise, StudentParticipation participation, Path outputDir, boolean includeResults,
+            Optional<List<String>> exportErrors) {
         Set<QuizQuestion> quizQuestions = quizQuestionRepository.getQuizQuestionsByExerciseId(quizExercise.getId());
-        QuizSubmission quizSubmission;
-
+        boolean errorOccurred = false;
         for (var submission : participation.getSubmissions()) {
-            quizSubmission = quizSubmissionRepository.findWithEagerSubmittedAnswersById(submission.getId());
+            QuizSubmission quizSubmission = quizSubmissionRepository.findWithEagerSubmittedAnswersById(submission.getId());
             List<String> multipleChoiceQuestionsSubmissions = new ArrayList<>();
             List<String> shortAnswerQuestionsSubmissions = new ArrayList<>();
             for (var question : quizQuestions) {
                 var submittedAnswer = quizSubmission.getSubmittedAnswerForQuestion(question);
                 // if this question wasn't answered, the submitted answer is null
                 if (submittedAnswer != null) {
                     if (submittedAnswer instanceof DragAndDropSubmittedAnswer dragAndDropSubmittedAnswer) {
-                        dragAndDropQuizAnswerConversionService.convertDragAndDropQuizAnswerAndStoreAsPdf(dragAndDropSubmittedAnswer, outputDir, includeResults);
+                        try {
+                            dragAndDropQuizAnswerConversionService.convertDragAndDropQuizAnswerAndStoreAsPdf(dragAndDropSubmittedAnswer, outputDir, includeResults);
+                        }
+                        catch (Exception e) {
+                            errorOccurred = true;
+                            exportErrors.ifPresent(errors -> errors.add("Failed to export drag and drop answers for quiz submission " + submission.getId() + " of quiz exercise "
```

Comment:
Why don't we include the error in a list instead of some string?

Reply:
this is a list, it's just wrapped in an optional to avoid the if (list != null) check